### DISCUSSION TITLE
How can I handle simultaneous interactions in Discord.py?

### DISCUSSION THREAD

**Discussion by Barefoot-Dev** [Rapptz/discord.py#9012] [General] [answered]
I'm setting up a discord.py Client like so: (from [here](https://github.com/Rapptz/discord.py/blob/master/examples/app_commands/basic.py))

```
class MyClient(discord.Client):
    def __init__(self, *, intents: discord.Intents):
        super().__init__(intents=intents)
        self.tree = app_commands.CommandTree(self)
```

In my main script, I then handle for interactions like this:
```

@client.tree.command()
async def test(
    interaction: discord.Interaction,
):
    print("Received interaction")

    # defer
    await interaction.response.defer()

    # sleep for 10 seconds
    print("Sleeping")
    time.sleep(10)

    # respond
    await interaction.followup.send("Test complete")
```

If I call the /test command twice in a row (within 10 seconds), the second one will immediately get a "The application did not respond" message, but after 10 seconds will still try to process, but then the interaction is Unknown and raises the following error:

`discord.errors.NotFound: 404 Not Found (error code: 10062): Unknown interaction`

I assume I need some sort of parallel processing, but I've tried awaiting a separate async function and also using a multiprocessing.Process to do the work (`sleep(10)`) and still hit the same issue.

**Comment by CDE90** [comment, marked as answer]
Have you tried using asyncio's sleep function? e.g.
```py 
await asyncio.sleep(10)
``` 
This should work without blocking execution globally
